src/libvirtApi/interface.js

```diff
@@ -1,75 +0,0 @@
-/*
- * This file is part of Cockpit.
- *
- * Copyright (C) 2021 Red Hat, Inc.
- *
- * Cockpit is free software; you can redistribute it and/or modify it
- * under the terms of the GNU Lesser General Public License as published by
- * the Free Software Foundation; either version 2.1 of the License, or
- * (at your option) any later version.
- *
- * Cockpit is distributed in the hope that it will be useful, but
- * WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
- * Lesser General Public License for more details.
- *
- * You should have received a copy of the GNU Lesser General Public License
- * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
- */
-
-/*
- * Provider for Libvirt using libvirt-dbus API.
- * See https://github.com/libvirt/libvirt-dbus
- */
-import store from '../store.js';
-
-import { updateOrAddInterface } from '../actions/store-actions.js';
-import { parseIfaceDumpxml } from '../libvirt-xml-parse.js';
-import { call, Enum, timeout } from './helpers.js';
-
-/*
- * Read properties of a single Interface
- *
- * @param {object} objPath interface object path
- * @param {string} connectionName
- */
-export async function interfaceGet({
-    id: objPath,
-    connectionName,
-}) {
-    const props = {};
-
-    try {
-        const [resultProps] = await call(connectionName, objPath, 'org.freedesktop.DBus.Properties', 'GetAll',
-                                         ['org.libvirt.Interface'], { timeout, type: 's' });
-        /* Sometimes not all properties are returned; for example when some network got deleted while part
-            * of the properties got fetched from libvirt. Make sure that there is check before reading the attributes.
-            */
-        if ("Active" in resultProps)
-            props.active = resultProps.Active.v.v;
-        if ("MAC" in resultProps)
-            props.mac = resultProps.MAC.v.v;
-        if ("Name" in resultProps)
-            props.name = resultProps.Name.v.v;
-        props.id = objPath;
-        props.connectionName = connectionName;
-
-        const [xml] = await call(connectionName, objPath, 'org.libvirt.Interface', 'GetXMLDesc', [0], { timeout, type: 'u' });
-        const iface = parseIfaceDumpxml(xml);
-        store.dispatch(updateOrAddInterface(Object.assign(props, iface)));
```

Comment:
This needs to be cleaned up as well then

https://github.com/search?q=repo%3Acockpit-project%2Fcockpit-machines%20UPDATE_ADD_INTERFACE&type=code

Reply:
And `updateOrAddInterface` is not called anymore with your PR, so I don't really understand how stuff in **this** PR works.